Add basic r11s end to end tests

jatgarg opened the pull request:
Issue: https://github.com/microsoft/FluidFramework/issues/3114

jatgarg committed:
add r11s end to end tests

jatgarg committed:
process

jatgarg committed:
Merge branch 'master' of https://github.com/microsoft/FluidFramework into r11stests

jatgarg committed:
add yaml

markfields reviewed (changes requested):
Jatin and I just chatted - This is exactly what I'm looking to add to run post-checkin via the Prod Test Harness, for both r11s and SPO. This shouldn't be merged as-is with the tests running with the end-to-end tests because then if r11s is down, no one can check in.  Jatin and I are going to work together to port this over to the Prod Test Harness.

jatgarg committed:
add yaml

jatgarg committed:
add yaml

jatgarg committed:
add yaml

jatgarg committed:
add yaml

jatgarg committed:
add yaml

markfields committed:
Move r11s tests to post-checkin folder

markfields committed:
Merge branch 'main' into r11stests

markfields committed:
Add post-build stage and for client-build run tests

markfields committed:
typo

markfields committed:
Trying to get variables syntax right

markfields committed:
Remove variables from build-client.yml

Not sure why I'm getting an error that variables is already defined. Maybe it's because the Azure Pipelines UX is inserting (secret) variables?

markfields committed:
Remove variables (for now) and add Install steps

Also placeholder for downloading artifacts

markfields committed:
Merge branch 'main' into r11stests

markfields commented:
Can't push more commits here due to repo changing visibility, I've pushed this branch to my new public fork and will submit a PR from there when ready.
